File: src/test/java/com/ddd/trainings/TrainingTests.java
package com.ddd.trainings;

import com.ddd.trainings.idea.api.app.TrainingIdeaService;
import com.ddd.trainings.idea.domain.TrainingIdeaBuilder;
import com.ddd.trainings.idea.infra.TrainingIdeaRepo;
import com.ddd.trainings.infra.InMemoryReviewerRepo;
import com.ddd.trainings.infra.InMemoryTrainingIdeaRepo;
import com.ddd.trainings.infra.InMemoryTrainingProposalRepo;
import com.ddd.trainings.proposal.domain.ReviewerPolicyFactory;
import com.ddd.trainings.proposal.infra.ReviewerRepo;
import com.ddd.trainings.proposal.infra.TrainingProposalRepo;
import org.junit.jupiter.api.BeforeEach;

public abstract class TrainingTests {

  protected TrainingIdeaService service;
  protected TrainingIdeaRepo ideaRepo;
  protected TrainingProposalRepo proposalRepo;
  protected ReviewerRepo reviewerRepo;
  protected TrainingIdeaBuilder trainingIdeaBuilder;

  @BeforeEach
  void setUp() {
    ideaRepo = new InMemoryTrainingIdeaRepo();
    proposalRepo = new InMemoryTrainingProposalRepo();
    reviewerRepo = new InMemoryReviewerRepo();
    service = new TrainingIdeaService(ideaRepo, proposalRepo, new ReviewerPolicyFactory(reviewerRepo));
    
    initBuilders();
  }

  protected void initBuilders() {
    trainingIdeaBuilder = new TrainingIdeaBuilder(ideaRepo);
  }
}

File: src/Ddd.Tests/Trainings/TrainingTests.cs
using Ddd.Trainings.Idea.Api.App;
using Ddd.Trainings.Idea.Domain.Tests;
using Ddd.Trainings.Idea.Domain.Ports;
using Ddd.Trainings.Proposal.Domain;
using Ddd.Trainings.Proposal.Domain.Ports;
using Ddd.Tests.Trainings.Idea.Infra.Tests;
using Ddd.Tests.Trainings.Proposal.Infra.Tests;

namespace Ddd.Tests.Trainings.Tests;

public abstract class TrainingTests
{
  protected TrainingIdeaService Service;
  protected ITrainingIdeaRepo IdeaRepo;
  protected ITrainingProposalRepo ProposalRepo;
  protected IReviewerRepo ReviewerRepo;
  protected TrainingIdeaBuilder TrainingIdeaBuilder;

  protected static readonly ReviewerPolicy FakeReviewerPolicy = new FakeReviewerPolicy();

  public TrainingTests()
  {
    IdeaRepo = new InMemoryTrainingIdeaRepo();
    ProposalRepo = new InMemoryTrainingProposalRepo();
    ReviewerRepo = new InMemoryReviewerRepo();
    Service = new TrainingIdeaService(IdeaRepo, ProposalRepo, new ReviewerPolicyFactory(ReviewerRepo));

    InitBuilders();
  }

  protected void InitBuilders()
  {
    TrainingIdeaBuilder = new TrainingIdeaBuilder(IdeaRepo);
  }
}

public class FakeReviewerPolicy : ReviewerPolicy
{
  public ReviewerId SelectReviewer()
  {
    return new ReviewerId("fake-reviewer-id");
  }
}